### ISSUE TITLE
Container build is failing for OpenSSL dependency

### ISSUE THREAD

**Issue by imavroukakis** [aws-samples/aws-iot-securetunneling-localproxy#134]
The build with the `latest` Amazon Linux container is failing with the following

```
CACHED [builder  1/13] FROM docker.io/library/amazonlinux:latest@sha256:1df62ceb869d3c8eef4dc48f944de3a27cbddcb16887c65
------
 > [stage-1 2/5] RUN yum check-update; yum upgrade -y &&     yum install -y openssl11 wget libatomic &&     rm -rf /var/cache/yum &&     yum clean all:
#5 8.101 Amazon Linux 2023 repository                    1.8 MB/s |  13 MB     00:07
#5 10.48 Last metadata expiration check: 0:00:05 ago on Tue May  9 14:16:09 2023.
#5 10.99 Last metadata expiration check: 0:00:05 ago on Tue May  9 14:16:09 2023.
#5 11.03 Dependencies resolved.
#5 11.03 Nothing to do.
#5 11.03 Complete!
#5 11.24 Last metadata expiration check: 0:00:05 ago on Tue May  9 14:16:09 2023.
#5 11.26 No match for argument: openssl11
#5 11.27 Error: Unable to find a match: openssl11
```

Looks like the last container tag to build correctly was `2.0.20230418.0`

**Comment by HarshGandhi-AWS** [comment]
Hello @imavroukakis, can you share the steps to reproduce this issue?

**Comment by imavroukakis** [comment]
Hi @HarshGandhi-AWS , I checked out the latest `main` and ran `./docker-build.sh`. 

```
 ./docker-build.sh
Architecture: x86_64
OpenSSL configurations: linux-generic64
[+] Building 20.2s (6/20)
 => [internal] load build definition from Dockerfile                                                                         0.1s
 => => transferring dockerfile: 3.30kB                                                                                       0.0s
 => [internal] load .dockerignore                                                                                            0.0s
 => => transferring context: 2B                                                                                              0.0s
 => [internal] load metadata for docker.io/library/amazonlinux:latest                                                        3.7s
 => CACHED [builder  1/13] FROM docker.io/library/amazonlinux:latest@sha256:1df62ceb869d3c8eef4dc48f944de3a27cbddcb16887c65  0.0s
 => CANCELED [builder  2/13] RUN yum check-update; yum upgrade -y &&  yum install -y git boost-devel autoconf automake  wg  16.3s
 => ERROR [stage-1 2/5] RUN yum check-update; yum upgrade -y &&     yum install -y openssl11 wget libatomic &&     rm -rf   16.2s
------
 > [stage-1 2/5] RUN yum check-update; yum upgrade -y &&     yum install -y openssl11 wget libatomic &&     rm -rf /var/cache/yum &&     yum clean all:
#5 9.200 Amazon Linux 2023 repository                    1.9 MB/s |  13 MB     00:06
#5 14.14 Last metadata expiration check: 0:00:07 ago on Fri May 12 07:18:31 2023.
#5 15.24 Last metadata expiration check: 0:00:08 ago on Fri May 12 07:18:31 2023.
#5 15.32 Dependencies resolved.
#5 15.33 Nothing to do.
#5 15.33 Complete!
#5 15.96 Last metadata expiration check: 0:00:08 ago on Fri May 12 07:18:31 2023.
#5 16.02 No match for argument: openssl11
#5 16.06 Error: Unable to find a match: openssl11
------
executor failed running [/bin/sh -c yum check-update; yum upgrade -y &&     yum install -y openssl11 wget libatomic &&     rm -rf /var/cache/yum &&     yum clean all]: exit code: 1
```

**Comment by HarshGandhi-AWS** [comment]
Thank you for sharing info. I was able to reproduce the issue. I will keep working on resolving the issue. 

Another thing I found was the method you are using to creating a docker image is [deprecated](https://github.com/aws-samples/aws-iot-securetunneling-localproxy#deprecated-method) so I would suggest following [these instructions](https://github.com/aws-samples/aws-iot-securetunneling-localproxy#deprecated-method) to build docker images & containers.
